### PR TITLE
validate-groups now ignores generated files

### PR DESCRIPTION
**Pull request by facchettos**
### Which issue this PR addresses:
Validate groups can conflict with generated files. This fixes it by ignoring generated files. 

Fixes

### CODE REVIEW

**Review by s-amann — APPROVED** [review]
lgtm

File: hack/validate-imports/validate-groups.go [hunk 1]
```diff
@@ -40,6 +42,15 @@ func typeForImport(imp *ast.ImportSpec) importType {
 }
 
 func validateGroups(path string, fset *token.FileSet, f *ast.File) (errs []error) {
+	b, err := os.ReadFile(path)
+	if err != nil {
+		errs = append(errs, err)
+		return
+	}
+	// some generated files are conflicting with this rule so we exclude those
+	if bytes.Contains(b, []byte("DO NOT EDIT.")) {
```
[the remark this answers is not in the document]
Comment (facchettos) [reply hack/validate-imports/validate-groups.go:51]: yes I agree, but it was done like that for the other validation , in `hack/validate-imports/validate-imports.go` so I didn't bother finding a better way ... I'm open to suggestions though if you know of a better way !

**Review by ross-bryan — APPROVED** [review]
lgtm

File: hack/validate-imports/validate-groups.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cadenmarchese) [reply hack/validate-imports/validate-groups.go:51]: Might be an easy way to harden it a little:
```suggestion
	if bytes.Contains(b, []byte("DO NOT EDIT.")) && bytes.Contains(b, []byte("Code generated")) {
```
All the generated code (I think) has both strings.

**Review by cadenmarchese — APPROVED** [review]

File: hack/validate-imports/validate-groups.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bennerv) [reply hack/validate-imports/validate-groups.go:51]: Would it make more sense to key off the mocks directory where everything is in?  

`pkg/util/mocks`

[the remark this answers is not in the document]
Comment (bennerv) [reply hack/validate-imports/validate-groups.go:51]:
```suggestion 

	if strings.Contains(path, "pkg/util/mocks") {

[the remark this answers is not in the document]
Comment (facchettos) [reply hack/validate-imports/validate-groups.go]: @bennerv follow your recommendation

**Review by SudoBrendan — APPROVED** [review]
LGTM

**Review by bennerv — APPROVED** [review]
lgtm

**Review by s-amann — APPROVED** [review]

**Comment by bennerv** [comment]
doesn't impact e2e at all.  merging.

**Pull request merged**